modules/rosetta/src/main/scala/org/tessellation/rosetta/domain/networkapi/NetworkApiService.scala

```diff
@@ -0,0 +1,62 @@
+package org.tessellation.rosetta.domain.networkapi
+
+import cats.data.EitherT
+import cats.effect.Async
+
+import org.tessellation.BuildInfo
+import org.tessellation.rosetta.domain.NetworkIdentifier
+import org.tessellation.rosetta.domain.error.{LatestSnapshotNotFound, NetworkApiError}
+import org.tessellation.rosetta.domain.networkapi.model.options._
+import org.tessellation.rosetta.domain.networkapi.model.status._
+import org.tessellation.schema.{GlobalSnapshot, SnapshotOrdinal}
+import org.tessellation.security.Hashed
+import org.tessellation.security.hex.Hex
+
+trait NetworkApiService[F[_]] {
+  def list: EitherT[F, NetworkApiError, List[NetworkIdentifier]]
+  def options: EitherT[F, NetworkApiError, NetworkApiOptions]
+  def status: EitherT[F, NetworkApiError, NetworkStatusResponse]
+}
+
+object NetworkApiService {
+  private val rosettaVersion = "1.4.14" // TODO get from a config?
+  private val version = Version(rosettaVersion = rosettaVersion, nodeVersion = BuildInfo.version)
+
+  def make[F[_]: Async](
+    getLastSnapshot: () => F[Option[Hashed[GlobalSnapshot]]],
+    getGenesisSnapshot: () => F[GlobalSnapshot],
+    getTimestampForSnapshot: SnapshotOrdinal => F[Long]
+  ): NetworkApiService[F] = new NetworkApiService[F] {
+    def list: EitherT[F, NetworkApiError, List[NetworkIdentifier]] =
+      EitherT.pure(List(NetworkIdentifier.Mainnet, NetworkIdentifier.Testnet))
+
+    def options: EitherT[F, NetworkApiError, NetworkApiOptions] =
+      EitherT.pure(NetworkApiOptions(version, Allow.default))
+
+    def status: EitherT[F, NetworkApiError, NetworkStatusResponse] = for {
+      lastSnapshot <- EitherT.fromOptionF[F, NetworkApiError, Hashed[GlobalSnapshot]](
+        getLastSnapshot(),
+        LatestSnapshotNotFound
+      )
+      genesisSnapshot <- EitherT.liftF(getGenesisSnapshot())
+      blockTimestamp <- EitherT.liftF(getTimestampForSnapshot(lastSnapshot.ordinal))
+
+      genesisBlockId = BlockIdentifier(genesisSnapshot.ordinal, Hex(genesisSnapshot.lastSnapshotHash.value))
+      currentBlockId = BlockIdentifier(lastSnapshot.ordinal, Hex(lastSnapshot.hash.value))
```

Comment:
why `currentBlockId` uses `lastSnapshot.hash` but `genesisBlockId` uses `genesisSnapshot.lastSnapshotHash`?

Reply:
`genesisSnapshot` is not a `Hashed[GlobalSnapshot]` so `.hash` is not defined.
Is `.lastSnapshotHash` not the valid hash to use?

Reply:
Since we're now injecting the function that gives us genesis I could change the return type so that it's the caller's responsibility to give me a `Hashed[GlobalSnapshot]` then the block ids would be set similarly.

Reply:
`hash` and `lastSnapshotHash` are two different things. while `hash` references to the current data, `lastSnapshotHash` references to previous data. IMO `genesisBlockId` should present genesis hash. genesis `lastSnapshotHash` is an empty hash (no reference).